### PR TITLE
Create and test MintableToken contract

### PR DESCRIPTION
**Pull request by DavidKnott**
Creates MintableToken contract that inherits from the StandardTokenContract.
MintableToken has an owner who is able to:

1) Mint tokens to a given account
2) Finish minting to let relevant parties know that minting is complete
3) Recover tokens from a given account

This contract was created in response to issue #120 
It's fully tested in MintableToken.js and all tests are passing.

### CODE REVIEW

File: contracts/token/MintableToken.sol [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  }
+
+  function mint(address _to, uint _amount) onlyOwner canMint returns (bool) {
+    totalSupply = safeAdd(totalSupply, _amount);
+    balances[_to] = safeAdd(balances[_to], _amount);
+    Transfer(0, _to, _amount);
+    return true;
+  }
+
+  function finishMinting() returns (bool) {
```
Comment (maraoz) [inline contracts/token/MintableToken.sol]: onlyOwner?

**Comment by DavidKnott** [comment]
@maraoz Thanks for catching that! I added the onlyOwner modifier to finishMinting().

File: contracts/token/MintableToken.sol [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+  modifier canMint() {
+    if(mintingFinished) throw;
+    _;
+  }
+
+  function mint(address _to, uint _amount) onlyOwner canMint returns (bool) {
+    totalSupply = safeAdd(totalSupply, _amount);
+    balances[_to] = safeAdd(balances[_to], _amount);
+    Transfer(0, _to, _amount);
```
Comment (maraoz) [inline contracts/token/MintableToken.sol]: Not sure about using Transfer here, I think the semantics are a little bit off. Should we create a new `Mint` event for this?

File: contracts/token/MintableToken.sol [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    return true;
+  }
+
+  function finishMinting() onlyOwner returns (bool) {
+    mintingFinished = true;
+    return true;
+  }
+
+  function recover(address _from) onlyOwner returns (bool) {
```
Comment (maraoz) [inline contracts/token/MintableToken.sol]: What's the idea of this method? Seems not related to the minting process. As such, should it be part of a separate contract? This seems to allow the owner to destroy arbitrary tokens, which doesn't seem to be related to the minting functionality.

File: contracts/token/MintableToken.sol (hunk 2 above)
Comment (DavidKnott) [inline contracts/token/MintableToken.sol]: I think you're right, my initial intention was to limit the amount of additional events to make searching for them easier but it's probable different enough from Transfer that it deserves its own event.

File: contracts/token/MintableToken.sol (hunk 3 above)
Comment (DavidKnott) [inline contracts/token/MintableToken.sol]: I added in the recover function because I think the owner should be able to destroy tokens from accounts that are lost or stolen.  Would you create a separate ```RecoverableToken``` contract inheriting from ```StandardToken``` or make it into a helper contract?

Comment (maraoz) [inline contracts/token/MintableToken.sol]: For now, I'd remove it from this contract, as it seems unrelated. 
For a `RecoverableToken` (please do it in another PR), I would do it inherit from `StandardToken`

**Comment by DavidKnott** [comment]
@maraoz I removed the recover function and created a separate ```Mint``` event.  Are there any other changes you see that need to be made?

**Comment by maraoz** [comment]
LGTM!

**Comment by tjayrush** [comment]
An event in finishedMinting would indicate to any process that may be monitoring the token that the tokens are all minted. Without such an event, said 'monitor' would have to remain alert to new minting. Is there a downside to additional events?

**Comment by maraoz** [comment]
@tjayrush No, that's a good idea. I opened an issue: https://github.com/OpenZeppelin/zeppelin-solidity/issues/191
